contributors/design-proposals/architecture/namespaces.md

```diff
@@ -191,6 +191,52 @@ If the *namespace controller* sees a *Namespace* whose
 list is empty, it will signal the server to permanently remove the *Namespace*
 from storage by sending a final DELETE action to the API server.
 
+There are situations where the *namespace controller* is unable to guarantee 
+cleanup of all resources. During a cleanup run, it attempts a best-effort 
+resource deletion, remembers the errors that occurred and report back via
+**namespace status condition**. Some errors can be transient and will
+auto-resolve in the following cleanup runs, others may require manual
+intervention.
+
+There are following status conditions reporting on the process of namespace
```

Comment:
This wording seems a little awkward - "There are following status conditions ...". Maybe it should be "There are the following status conditions ..." or "These are the status conditions ..."

Reply:
I like the second wording you have mentioned very much - changing to `These are the status conditions ...`